db/dbScript.sql

```diff
@@ -0,0 +1,393 @@
+CREATE EXTENSION IF NOT EXISTS "uuid-ossp"; --it is needed for uuid generator (uuid_generate_v4()) to be available
+
+--Basically, we dynamically create enums if they do not exist
+BEGIN
+    IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'provider_type') THEN
+        CREATE TYPE provider_type AS ENUM ('local', 'google');
+    END IF;
+	IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'account_type') THEN
+        CREATE TYPE account_type AS ENUM ('user', 'gym', 'admin');
+    END IF;
+	IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'own_status') THEN
+        CREATE TYPE own_status AS ENUM ('pending', 'approved', 'rejected');
+    END IF;
+	IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'rec_type') THEN
+        CREATE TYPE rec_type AS ENUM ('main', 'alternative');
+    END IF;
+	IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'not_type') THEN
+        CREATE TYPE not_type AS ENUM ('message', 'alert', 'reminder');
+    END IF;
+END $$;
+
+--Since we will have a lot of gyms stored (potentially) it does not make sense to store working hours for each of them as a text
+--Especially since there are significantly smaller potential patterns [(opening hours, closing hours) pairs] in comparison to the potential number of gyms stored
+CREATE TABLE if not exists Working_hours
+(
+  id uuid NOT null DEFAULT uuid_generate_v4(),--uuid is analogous to id, but uuid is pseudo-random and the length and format of uuids are the same
```

Comment:
I’m not sure that using UUID is necessary for this table since it functions as a simple list of values. However, we might keep it for consistency with more complex tables in the database.

Reply:
Basically, here we have just time ranges, so not even working days or anything. Then we have a table Gym_working_hours where we map gyms to the Working_hours we have created in this table for each of the working days

For instance, we have working hours in Working_hours table from 10 to 20 with id equal to some uuid1
Then for a certain gym that works from 10 to 20 each day of the weak, we will have the following entries in the Gym_working_hours db
|gym_id| working_hour id|cardinal number of the week |
|--------|--------|-------|
|gym_id |uuid1| 0|
|gym_id |uuid1 |1|
|gym_id| uuid1 |2|
|...|||
|gym_id| uuid1| 6|

That was my idea